source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2478 @@
+
+[libmongocrypt](#libmongocrypt)\
+A library, written in C, that coordinates communication, does encryption/decryption,
+caches key and schemas. [Located here](https://github.com/mongodb/libmongocrypt).
+
+[mongocryptd](#mongocryptd)\
+A local process the driver communicates with to determine how to encrypt values in a
+command.
+
+[crypt_shared](#crypt_shared)\
+This term, spelled in all-lowercase with an underscore, refers to the client-side
+field-level-encryption dynamic library provided as part of a MongoDB Enterprise distribution. It replaces
+[mongocryptd](#mongocryptd) as the method of
+`marking-up a database command for encryption <subtype6.intent-to-encrypt>`.
+
+See also:
+
+> - [Introduction on crypt_shared](#crypt_shared)
+> - \[Enabling crypt_shared\](#Enabling crypt_shared)
+
+ciphertext\
+One of the data formats of
+[BSON binary subtype 6](https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/subtype6.rst),
+representing an encoded BSON document containing encrypted ciphertext and metadata.
+
+FLE\
+FLE is the first version of Client-Side Field Level Encryption. FLE is almost entirely client-side with the
+exception of server-side JSON schema.
+
+Queryable Encryption\
+Queryable Encryption the second version of Client-Side Field Level Encryption. Data is encrypted
+client-side. Queryable Encryption supports indexed encrypted fields, which are further processed server-side.
+
+In-Use Encryption\
+Is an umbrella term describing the both FLE and Queryable Encryption.
+
+encryptedFields\
+A BSON document describing the Queryable Encryption encrypted fields. This is analogous to the JSON
+Schema in FLE. The following is an example encryptedFields in extended canonical JSON:
+
+```
+{
+    "escCollection": "enxcol_.CollectionName.esc",
+    "ecocCollection": "enxcol_.CollectionName.ecoc",
+    "fields": [
+        {
+            "path": "firstName",
+            "keyId": { "$binary": { "subType": "04", "base64": "AAAAAAAAAAAAAAAAAAAAAA==" }},
+            "bsonType": "string",
+            "queries": {"queryType": "equality"}
+        },
+        {
+            "path": "ssn",
+            "keyId": { "$binary": { "subType": "04", "base64": "BBBBBBBBBBBBBBBBBBBBBB==" }},
+            "bsonType": "string"
+        }
+    ]
+}
+```
+
+The acronyms within `encryptedFields` are defined as follows:
+
+- ECOC: Encrypted Compaction Collection
+- ESC: Encrypted State Collection
+
+## Introduction
+
+Client side encryption enables users to specify what fields in a collection must be encrypted, and the driver
+automatically encrypts commands and decrypts results. Automatic encryption is enterprise only. But users can manually
+encrypt and decrypt with a new ClientEncryption object.
+
+Client side encryption requires MongoDB 4.2 compatible drivers, and is only supported against 4.2 or higher servers. See
+[Why is a 4.2 server required?](#why-is-a-4.2-server-required).
```

Comment:
```suggestion
[Why is a 4.2 server required?](#why-is-a-42-server-required).
```

The Markdown slug rules appear to be different from RST, so I think a lot of these links are probably incorrect. I'll let you go through these and follow up on it in a subsequent review.

Reply:
Updated